airflow/providers/google/cloud/hooks/datafusion.py

```diff
@@ -435,20 +435,52 @@ def _get_pipeline(
             "workflows",
             "DataPipelineWorkflow",
             "runs",
+            pipeline_id,
         )
         response = self._cdap_request(url=url, method="GET")
         if response.status != 200:
             raise AirflowException(
-                f"Retrieving a pipeline failed with code {response.status}"
+                f"Retrieving a pipeline state failed with code {response.status}"
             )
+        workflow = json.loads(response.data)
+        return workflow["status"]
 
-        pipelines_list = json.loads(response.data)
-        for pipe in pipelines_list:
-            runtime_args = json.loads(pipe["properties"]["runtimeArgs"])
-            if runtime_args[job_id_key] == faux_pipeline_id:
-                return pipe
+    def _get_pipeline_run_id(
+        self,
+        pipeline_name: str,
+        faux_pipeline_id: str,
+        instance_url: str,
+        namespace: str = "default",
+    ) -> str:
+        url = os.path.join(
+            instance_url,
+            "v3",
+            "namespaces",
+            namespace,
+            "apps",
+            pipeline_name,
+            "workflows",
```

Comment:
This CDAP API is very convoluted.

Seems like there are [several program types](https://docs.cask.co/cdap/6.1.1/en/reference-manual/http-restful-api/lifecycle.html#H3581) and this is hard coding `workflows` and `DataPipelineWorkflow` .

I think `DataPipelineWorkflow` this will not be present for streaming pipelines instead you have to poll a `spark` program.

I'm not sure how many other scenarios require other program types.
It would be good to get someone from CDAP community to review this.

![Screenshot 2020-05-22 at 10 21 00 AM](https://user-images.githubusercontent.com/11599048/82693211-f791df80-9c15-11ea-964b-dc8fcad3e25c.png)

![Screenshot 2020-05-22 at 10 20 27 AM](https://user-images.githubusercontent.com/11599048/82693231-037da180-9c16-11ea-8a0d-c6665c416f2a.png)

Reply:
CC: @sreevatsanraman

Reply:
This seems to be a bigger change because we will have to adjust each method that uses `DataPipelineWorkflow` in URI. So, I would say we can do this but in a follow up PR.

Btw. I was basing on Google docs: https://cloud.google.com/data-fusion/docs/reference/cdap-reference#start_a_batch_pipeline

Reply:
As long as we cover batch pipelines (with spark or MR backend I think we should be good)